pkg/controllers/rollout/controller.go

```diff
@@ -548,3 +613,52 @@ func handleResourceBinding(binding client.Object, q workqueue.RateLimitingInterf
 		NamespacedName: types.NamespacedName{Name: crp},
 	})
 }
+
+// updateStaleBindingsStatus updates the status of the stale bindings to indicate that they are blocked by the rollout strategy.
+// Note: the binding state should be "Scheduled" or "Bound".
+// The desired binding will be ignored.
+func (r *Reconciler) updateStaleBindingsStatus(ctx context.Context, staleBindings []toBeUpdatedBinding) error {
+	if len(staleBindings) == 0 {
+		return nil
+	}
+	// issue all the update requests in parallel
+	errs, cctx := errgroup.WithContext(ctx)
+	for i := 0; i < len(staleBindings); i++ {
+		binding := staleBindings[i]
+		if binding.currentBinding.Spec.State != fleetv1beta1.BindingStateScheduled && binding.currentBinding.Spec.State != fleetv1beta1.BindingStateBound {
+			klog.ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("invalid stale binding state %s", binding.currentBinding.Spec.State)),
+				"Found a stale binding with unexpected state", "clusterResourceBinding", klog.KObj(binding.currentBinding))
+			continue
+		}
+		errs.Go(func() error {
+			return r.updateBindingStatus(cctx, binding.currentBinding, false)
+		})
+	}
+	return errs.Wait()
+}
+
+func (r *Reconciler) updateBindingStatus(ctx context.Context, binding *fleetv1beta1.ClusterResourceBinding, rolloutStarted bool) error {
+	cond := metav1.Condition{
+		Type:               string(fleetv1beta1.ResourceBindingRolloutStarted),
+		Status:             metav1.ConditionFalse,
+		ObservedGeneration: binding.Generation,
+		Reason:             condition.RolloutNotStartedYetReason,
+		Message:            "The resources cannot be updated to the latest because of the rollout strategy",
+	}
+	if rolloutStarted {
+		cond = metav1.Condition{
+			Type:               string(fleetv1beta1.ResourceBindingRolloutStarted),
+			Status:             metav1.ConditionTrue,
+			ObservedGeneration: binding.Generation,
+			Reason:             condition.RolloutStartedReason,
+			Message:            "Updated the resource to the latest one and starting rolling out",
```

Comment:
Just a nit: the `starting rollout out` part might be a little bit confusing as this condition might persist for a long while.